File: ruby/easy/search_insert_position.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer}
def search_insert(nums, target)
    return bs(nums, 0, nums.length-1, target)
end

def bs(nums, start, stop, target)
    return stop+1 if start > stop
    middle = ((start + stop)/2).to_i
    if(nums[middle] == target)
        return middle
    elsif nums[middle] > target
        return bs(nums, start, middle-1, target)
    else
        return bs(nums, middle+1, stop, target)
    end
end

File: javascript/easy/searchInsertPosition.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var searchInsert = function(nums, target) {
    let start = 0, end = nums.length-1;
    while(start <= end) {
        let middle = parseInt((start+end)/2);
        if(nums[middle] == target) {
            return middle;
        } else if(nums[middle] > target) {
            end = middle-1;
        } else {
            start = middle+1;
        }
    }
    return end+1;
};
